lib/govuk_index/popularity_updater.rb

```diff
@@ -0,0 +1,93 @@
+module GovukIndex
+  class PopularityUpdater
+    SCROLL_BATCH_SIZE = 500
+    PROCESSOR_BATCH_SIZE = 100
+    MAX_SIDEKIQ_QUEUE_SIZE = 10
+    MAX_SIDEKIQ_WAIT = 5 * 60
+
+    class SidekiqWaitTimeoutExceeded < StandardError; end
+
+    def self.update(index_name)
+      index_group = SearchConfig.instance.search_server.index_group(index_name)
+      new_index = index_group.create_index
+      index_group.current.with_lock do
+        new(
+          source_index: index_group.current.real_name,
+          destination_index: new_index.real_name
+        ).run
+        index_group.switch_to(new_index)
+      end
+    end
+
+    def initialize(source_index:, destination_index:)
+      @source_index = source_index
+      @destination_index = destination_index
+    end
+
+    def run
+      scroll_enumerator.each_slice(PROCESSOR_BATCH_SIZE) do |documents|
+        wait_for_sidekiq
+        PopularityWorker.perform_async(documents, @destination_index)
+      end
+    end
+
+  private
+
+    def wait_for_sidekiq
```

Comment:
How does sidekiq behave when we publish 1000 of these messages without this throttling?

Could you leave a comment explaining the rationale for it?